generator/nanopb_generator.py

```diff
@@ -2049,7 +2055,7 @@ def main_plugin():
 
 if __name__ == '__main__':
     # Check if we are running as a plugin under protoc
-    if 'protoc-gen-' in sys.argv[0] or '--protoc-plugin' in sys.argv:
+    if 'protoc-gen-' in sys.argv[0] or '--protoc-plugin' in sys.argv or '_bazel_' in sys.argv[0]:
```

Comment:
Can you elaborate on how bazel is invoking `nanopb_generator.py` that it needs this special mechanism?

Reply:
This is simply a result of me misunderstanding your documentation. I had not spotted the shell/batch files that add the --protoc-plugin command line flags. For some reason I thought you where directly passing the nanopb_generator.py to protoc as a plugin. I can work around this by adding specific shell scripts for bazel. Or alternatively I could potentially split nanopb_generator.py into three smaller python files, nanopb_plugin.py, nanopb_cli.py, nanopb_lib.py. This would remove the need for proxy scripts altogether. Totally up to you, it's really personal preference at this point.

Reply:
TLDR; it's not needed, I can do better.

Reply:
Splitting nanopb_generator.py is something I've considered myself to improve maintainability, but better to do that separately eventually.

Using either the flag directly or the shell scripts should be fine. Because of Python's portability, just giving `--protoc-plugin` to nanopb_generator.py would avoid dependency on any particular shell on the build machine. It is also possible to just use `nanopb_generator.py myfile.proto`, in which case nanopb itself calls protoc behind the scenes.